test/jdk/tools/jpackage/macosx/SigningRuntimeImagePackageTest.java

```diff
@@ -0,0 +1,263 @@
+                "--type", "dmg",
+                "--name", "foo",
+                "--runtime-image", runtimeImageImage.toAbsolutePath().toString(),
+                "--dest", runtimeBundleDMG.toAbsolutePath().toString());
+
+            if (isRuntimeImageSigned) {
+                ex.addArguments(
+                    "--mac-sign",
+                    "--mac-signing-keychain", SigningBase.getKeyChain(),
+                    "--mac-signing-key-user-name", SigningBase.getDevName(JDKBundleCertIndex));
+            }
+
+            ex.execute();
+
+            JPackageCommand dummyCMD = new JPackageCommand();
+            dummyCMD.addArguments(
+                "--type", "dmg",
+                "--name", "foo",
+                "--dest", runtimeBundleDMG.toAbsolutePath().toString()
+                );
+
+            MacHelper.withExplodedDmg(dummyCMD, dmgImage -> {
+                if (dmgImage.endsWith(dummyCMD.name() + ".jdk")) {
+                    Executor.of("cp", "-R")
+                            .addArgument(dmgImage)
+                            .addArgument(runtimeBundleBundle.toAbsolutePath().toString())
+                            .execute(0);
+                }
+            });
+
+            return runtimeBundleBundle.toAbsolutePath();
+        } else {
+            return runtimeImageImage.toAbsolutePath();
+        }
+    }
+
+    @Test
+    // useJDKBundle  - If "true" predefined runtime image will be converted to
+    //                 JDK bundle. If "false" JDK image will be used.
+    // JDKBundleCert - Certificate to sign JDK bundle before calling jpackage.
+    // signCert      - Certificate to sign bundle produced by jpackage.
+    // 1) unsigned JDK bundle and --mac-sign is not provided
+    @Parameter({"true", "INVALID_INDEX", "INVALID_INDEX"})
+    // 2) unsigned JDK bundle and --mac-sign is provided
+    @Parameter({"true", "INVALID_INDEX", "ASCII_INDEX"})
+    // 3) signed JDK bundle and --mac-sign is not provided
+    @Parameter({"true", "UNICODE_INDEX", "INVALID_INDEX"})
+    // 4) signed JDK bundle and --mac-sign is provided
+    @Parameter({"true", "UNICODE_INDEX", "ASCII_INDEX"})
+    // 5) JDK image and --mac-sign is not provided
+    @Parameter({"false", "INVALID_INDEX", "INVALID_INDEX"})
+    // 6) JDK image and --mac-sign is provided
+    @Parameter({"false", "INVALID_INDEX", "ASCII_INDEX"})
+    public static void test(boolean useJDKBundle,
+                            SigningBase.CertIndex JDKBundleCert,
+                            SigningBase.CertIndex signCert) throws Exception {
+        final int JDKBundleCertIndex = JDKBundleCert.value();
+        final int signCertIndex = signCert.value();
+
+        final boolean isRuntimeImageSigned =
+            (JDKBundleCertIndex != SigningBase.CertIndex.INVALID_INDEX.value());
+        final boolean isSigned =
+            (signCertIndex != SigningBase.CertIndex.INVALID_INDEX.value());
+
+        new PackageTest()
+                .forTypes(PackageType.MAC)
```

Comment:
`.forTypes(PackageType.MAC)` is redundant. The test will only be executed on macOS